app/controllers/work_packages/moves_controller.rb

```diff
@@ -40,45 +40,53 @@ def new
   def create
     prepare_for_work_package_move
 
-    result = modify_call
-
-    set_flash_from_bulk_work_package_save(@work_packages, result)
-
-    redirect_after_create(result)
+    perform_operation
   end
 
   private
 
-  def modify_call
-    klass = if @copy
-              WorkPackages::Bulk::CopyService
-            else
-              WorkPackages::Bulk::MoveService
-            end
-
-    klass
-      .new(user: current_user, work_packages: @work_packages)
-      .call(attributes_for_create)
+  def perform_operation
+    if @work_packages.count <= Setting.work_packages_bulk_request_limit
```

Comment:
I think this is only the selected work package to be copied. This count would not include all the descendants, would it? In In the case that caused the bug report only a single work package had been selected to be copied. But it had 1000+ descendants.

Reply:
Good point, fixed